src/vanilla/internals.ts

```diff
@@ -155,33 +158,34 @@ type Store = {
-  unmountCallbacks: Callbacks, //                            5
-  storeHooks: StoreHooks, //                                 6
+  atomStateMap: AtomStateMap, //                               0
+  mountedMap: MountedMap, //                                   1
+  invalidatedAtoms: InvalidatedAtoms, //                       2
+  changedAtoms: ChangedAtoms, //                               3
+  mountCallbacks: Callbacks, //                                4
+  unmountCallbacks: Callbacks, //                              5
+  storeHooks: StoreHooks, //                                   6
   // atom interceptors
-  atomRead: AtomRead, //                                     7
-  atomWrite: AtomWrite, //                                   8
-  atomOnInit: AtomOnInit, //                                 9
-  atomOnMount: AtomOnMount, //                               10
+  atomRead: AtomRead, //                                       7
+  atomWrite: AtomWrite, //                                     8
+  atomOnInit: AtomOnInit, //                                   9
+  atomOnMount: AtomOnMount, //                                 10
   // building-block functions
-  ensureAtomState: EnsureAtomState, //                       11
-  flushCallbacks: FlushCallbacks, //                         12
-  recomputeInvalidatedAtoms: RecomputeInvalidatedAtoms, //   13
-  readAtomState: ReadAtomState, //                           14
-  invalidateDependents: InvalidateDependents, //             15
-  writeAtomState: WriteAtomState, //                         16
-  mountDependencies: MountDependencies, //                   17
-  mountAtom: MountAtom, //                                   18
-  unmountAtom: UnmountAtom, //                               19
-  setAtomStateValueOrPromise: SetAtomStateValueOrPromise, // 20
+  ensureAtomState: EnsureAtomState, //                         11
+  flushCallbacks: FlushCallbacks, //                           12
+  recomputeInvalidatedAtoms: RecomputeInvalidatedAtoms, //     13
+  readAtomState: ReadAtomState, //                             14
+  invalidateDependents: InvalidateDependents, //               15
+  writeAtomState: WriteAtomState, //                           16
+  mountDependencies: MountDependencies, //                     17
+  mountAtom: MountAtom, //                                     18
+  unmountAtom: UnmountAtom, //                                 19
+  setAtomStateValueOrPromise: SetAtomStateValueOrPromise, //   20
   // store api
-  storeGet: StoreGet, //                                     21
-  storeSet: StoreSet, //                                     22
-  storeSub: StoreSub, //                                     23
+  storeGet: StoreGet, //                                       21
+  storeSet: StoreSet, //                                       22
+  storeSub: StoreSub, //                                       23
+  enhanceBuildingBlocks: EnhanceBuildingBlocks | undefined, // 24
```

Comment:
I wonder if this can be something like this:
```suggestion
  alternateBuildingBlocks: Readonly<BuildingBlocks> | undefined, // 24
```

(I'm not sure if "alternate" is the right word. "real"? "overriding"?)

Reply:
```ts
const buildingBlocks = [new WeakMap()]
buildingBlocks[24] = [
  // ScopeProvider needs access to
  // all the internal building blocks
  // here so that it can shim them
  // but they are created internal
  // to buildStore and not exposed
]
const store = buildStore(...buildingBlocks)
```